### PR TITLE
Rename properties on base class Editor

### PR DESCRIPTION
**Pull request by gandarez**
- Make base class having properties `key` and `name`

### CODE REVIEW

File: src/editors/editor.ts [hunk 1]
```diff
@@ -1,6 +1,24 @@
...
+  abstract isEditorInstalled(): Promise<boolean>;
+
+  abstract isPluginInstalled(): Promise<boolean>;
+
+  abstract installPlugin(): Promise<void>;
+
+  abstract uninstallPlugin(): Promise<void>;
+}
+
+declare interface IEditor {
```
Comment (alanhamlett) [inline src/editors/editor.ts]: How about `EditorInterface`?

File: src/editors/vim.ts [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
-  public get displayName(): string {
     return "Vim";
   }
 
   public get icon(): string {
     return "";
   }
 
+  public get binary(): string {
+    return "vim";
```
Comment (alanhamlett) [inline src/editors/vim.ts]: Let's return an array of strings here, to support editors with multiple possible binary names:

```
public get binary(): string[] {
    return ['vi', vim'];
}
```

File: src/editors/vim.ts [hunk 3]
```diff
@@ -1,24 +1,24 @@
...
+  }
+
   public async isEditorInstalled(): Promise<boolean> {
     // @ts-ignore
-    return this.commandExists.exists(this.name);
+    return this.commandExists.exists(this.binary);
```
Comment (alanhamlett) [inline src/editors/vim.ts]: Then here we loop over binaries and return true if any of the binaries exist in PATH.

**Comment by alanhamlett** [comment]
The `binaries` array change can be done later, just the `package-lock.json` change is needed before merging.

File: src/editors/editor.ts (hunk 1 above)
Comment (gandarez) [reply src/editors/editor.ts]: It's most common to prefix with `I` for interfaces. Do you still prefer to add the suffix `Interface`?

**Comment by gandarez** [comment]
Already removed `package-lock.json` @alanhamlett

File: src/editors/editor.ts (hunk 1 above)
Comment (alanhamlett) [reply src/editors/editor.ts]: That's a C# convention, but TypeScript follows JavaScript conventions.

Comment (gandarez) [reply src/editors/editor.ts]: done!

File: src/editors/vim.ts (hunk 2 above)
Comment (gandarez) [reply src/editors/vim.ts]: done

File: src/editors/vim.ts (hunk 3 above)
Comment (gandarez) [reply src/editors/vim.ts]: done

**Comment by gandarez** [comment]
Renamed IEditor to EditorInterface and also make all `binary` methods to return an array of string.

**Pull request merged**